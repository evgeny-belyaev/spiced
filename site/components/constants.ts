import { isServer } from "../api/utils"

export const BaseUrl =
    process.env["NODE_ENV"] == "development" || process.env["NODE_ENV"] == "test" ?
        "http://localhost:5000/" :
        "https://spiced-f9677.web.app/"

export const Forms = {
    accessToken: isServer() ? "CnLWx37r88w9NoqC5H3UrkVM1UDgNYVBt9VQi22aDURm" : "",
    createCommunityFormId: "NaV9AthP"
}

export const MailChimp = {
    marketingAccessToken: isServer() ? "5ef474ca75d706884cc1b905dd8cf091-us2" : "",
    transactionalAccessToken : isServer() ? "AOOq73pypJb5533oSKdqEw" : "",
    server: isServer() ? "us2" : "",
    audienceId: isServer() ? "94657a4562" : ""
}

export const Database = {
    accessToken: isServer() ? "AIzaSyAsoRfHyVtfB0VE-t3eSfgpoEi0FLsXnH0" : "",
    url: "https://spiced-f9677.firebaseio.com/"
}