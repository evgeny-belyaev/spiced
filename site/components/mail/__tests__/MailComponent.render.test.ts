import { MailComponent } from "../index"
import { MailChimp } from "../../constants"

jest.unmock("axios")

export default describe("MailComponent", () => {
    test("renderTemplate: matched", async () => {
        // Arrange
        const mail = new MailComponent()
        const mailTemplate = MailChimp.Templates.matched

        // Act
        const result = await mail.renderTemplate(mailTemplate.name, [
            {
                name: mailTemplate.fields.matchedUserName,
                content: "matchedUserName"
            },
            {
                name: mailTemplate.fields.matchedUserEmail,
                content: "matchedUserEmail"
            },
            {
                name: mailTemplate.fields.matchedUserPhone,
                content: "matchedUserPhone"
            },
            {
                name: mailTemplate.fields.communityTitle,
                content: "communityTitle"
            }
        ])

        // Assert
        expect(result.html).toMatchSnapshot()
    })

    test("renderTemplate: communityCreated", async () => {
        // Arrange
        const mail = new MailComponent()
        const mailTemplate = MailChimp.Templates.communityCreated

        // Act
        const result = await mail.renderTemplate(mailTemplate.name, [
            {
                name: mailTemplate.fields.communityInvitationLink,
                content: "communityInvitationLink"
            },
            {
                name: mailTemplate.fields.communityTitle,
                content: "communityTitle"
            }
        ])

        // Assert
        expect(result.html).toMatchSnapshot()
    })

    test("renderTemplate: communityJoined", async () => {
        // Arrange
        const mail = new MailComponent()
        const mailTemplate = MailChimp.Templates.communityJoined

        // Act
        const result = await mail.renderTemplate(mailTemplate.name, [
            {
                name: mailTemplate.fields.communityTitle,
                content: "communityTitle"
            }
        ])

        // Assert
        expect(result.html).toMatchSnapshot()
    })

    test("renderTemplate: joinCommunityConfirmation", async () => {
        // Arrange
        const mail = new MailComponent()
        const mailTemplate = MailChimp.Templates.joinCommunityConfirmation

        // Act
        const result = await mail.renderTemplate(mailTemplate.name, [
            {
                name: mailTemplate.fields.joinCommunityConfirmationUrl,
                content: "joinCommunityConfirmationUrl"
            }
        ])

        // Assert
        expect(result.html).toMatchSnapshot()
    })

    test("renderTemplate: createCommunityConfirmation", async () => {
        // Arrange
        const mail = new MailComponent()
        const mailTemplate = MailChimp.Templates.createCommunityConfirmation

        // Act
        const result = await mail.renderTemplate(mailTemplate.name, [
            {
                name: mailTemplate.fields.createCommunityConfirmationUrl,
                content: "createCommunityConfirmationUrl"
            }
        ])

        // Assert
        expect(result.html).toMatchSnapshot()
    })

    test("renderTemplate: optIn", async () => {
        // Arrange
        const mail = new MailComponent()
        const mailTemplate = MailChimp.Templates.optIn

        // Act
        const result = await mail.renderTemplate(mailTemplate.name, [
            {
                name: mailTemplate.fields.communityTitle,
                content: "communityTitle"
            },
            {
                name: mailTemplate.fields.yesUrl,
                content: "<a href=\"yesUrl\">Yes</a>"
            },
            {
                name: mailTemplate.fields.noUrl,
                content: "<a href=\"noUrl\">No</a>"
            }
        ])

        // Assert
        expect(result.html).toMatchSnapshot()
    })
})
