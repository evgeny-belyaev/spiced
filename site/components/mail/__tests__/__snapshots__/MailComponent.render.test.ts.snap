// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MailComponent renderTemplate: communityCreated 1`] = `
"<div>
          Wow! You've created community: <span>communityTitle</span>
      </div>    
      <div>
      Here is invitation link: <span><a href=\\"https://a.b.com/\\">https://a.b.com/</a></span>
    </div>"
`;

exports[`MailComponent renderTemplate: communityJoined 1`] = `
"<div>
          Wow! You've joined to: <span>communityTitle</span>
      </div>"
`;

exports[`MailComponent renderTemplate: createCommunityConfirmation 1`] = `
"<div>Here is community confirmation url: <span><a href=\\"https://a.b.com/\\">https://a.b.com/</a></span>
    </div>"
`;

exports[`MailComponent renderTemplate: joinCommunityConfirmation 1`] = `
"<div>
    To join the community please follow this link : <span><a href=\\"https://a.b.com/\\">https://a.b.com/</a></span></div>"
`;

exports[`MailComponent renderTemplate: matched 1`] = `
"<h2>communityTitle</h2>
<div>
    You've matched with <span>matchedUserName</span></div>
    
<div>
    Contact email: <span>matchedUserEmail</span></div>
    
<div>
    Contact phone: <span>matchedUserPhone</span></div>"
`;

exports[`MailComponent renderTemplate: optIn 1`] = `
"<h2>communityTitle</h2>
<div>
    Would you like to match in <span>communityTitle</span></div>
    
<div>
    Click here to answer yes: <span><a href=\\"https://a.b.com/yes\\">https://a.b.com/yes</a></span></div>
    
<div>
    Click here to answer no: <span><a href=\\"https://a.b.com/no\\">https://a.b.com/no</a></span></div>"
`;
